File: frontend/lib/types/subscription.ts
export interface SubscriptionStatus {
  has_subscription: boolean;
  status: string | null;
  plan: string | null;
  credits_balance: number;
  credits_total: number;
  is_trial: boolean;
  trial_ends_at: string | null;
  current_period_end: string | null;
  canceled_at: string | null;
}

export interface CreditTransaction {
  id: string;
  type: string;
  amount: number;
  balance_after: number;
  operation_type: string | null;
  description: string | null;
  created_at: string;
}

export const CREDIT_COSTS: Record<string, number> = {
  video_generation_standard: 25,
  video_generation_fast: 10,
  video_extension_standard: 25,
  video_extension_fast: 10,
  face_analysis: 5,
  prompt_enhancement: 0,
};

File: backend/app/schemas/subscription.py
from pydantic import BaseModel
from typing import Optional, List
from uuid import UUID
from datetime import datetime


class SubscriptionStatusResponse(BaseModel):
    has_subscription: bool
    status: Optional[str] = None
    plan: Optional[str] = None
    credits_balance: int = 0
    credits_total: int = 0
    is_trial: bool = False
    trial_ends_at: Optional[datetime] = None
    current_period_end: Optional[datetime] = None
    canceled_at: Optional[datetime] = None


class CheckoutResponse(BaseModel):
    checkout_url: str
    checkout_id: str


class PortalResponse(BaseModel):
    portal_url: str


class CreditTransactionResponse(BaseModel):
    id: UUID
    type: str
    amount: int
    balance_after: int
    operation_type: Optional[str] = None
    description: Optional[str] = None
    created_at: datetime

    class Config:
        from_attributes = True


class CreditCostInfo(BaseModel):
    operation: str
    credits: int
